.github/workflows/ci.yaml

```diff
@@ -0,0 +1,120 @@
+name: Zipline CI
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build-and-test:
+    name: build (${{ matrix.os }}, ${{ matrix.python-version }}, ${{ matrix.pandas-version }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]  # TODO: Add windows-latest, macos-latest
+        python-version: [2.7, 3.5]
+        pandas-version: [0.18.1, 0.22.0]
+        include:  # TODO: Add 3.7, 3.8
+          - python-version: 2.7  # py2.7, old pandas
+            pandas-version: 0.18.1
+            numpy-version: 1.11.3
+            scipy-version: 0.17.1
+          - python-version: 3.5  # py3.5, new pandas
+            pandas-version: 0.18.1
+            numpy-version: 1.11.3
+            scipy-version: 0.17.1
+          - python-version: 3.5  # py3.5, new pandas
+            pandas-version: 0.22.0
+            numpy-version: 1.14.1
+            scipy-version: 1.0.0
+        exclude:
+          - python-version: 2.7
+            pandas-version: 0.22.0
+    env:
+      PYTHON_VERSION: ${{ matrix.python-version }}
+      PANDAS_VERSION: ${{ matrix.pandas-version }}
+      NUMPY_VERSION: ${{ matrix.numpy-version }}
+      SCIPY_VERSION: ${{ matrix.scipy-version }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Pip cache
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/.pip/pip_np$NUMPY_VERSION
```

Comment:
Given that we're using a `key`, I don't think we should need to encode the `numpy_version` into the pip cache directory anymore.